src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -900,6 +932,144 @@ public bool TryGetDeclaredSymbolInfo(SyntaxNode node, out DeclaredSymbolInfo dec
+        /// user provided.  That means that if they wrote "Method(System.Int32 i)" we'll 
+        /// show that as "Method(System.Int32)" not "Method(int)".  Given that this is
+        /// actually what the user wrote, and it saves us from ever having to go back to
+        /// symbols/compilations, this is well worth it, even if it does mean we have to
+        /// create our own 'symbol display' logic here.
+        /// </summary>
+        private string GetSuffix(
+            char openBrace, char closeBrace, SeparatedSyntaxList<ParameterSyntax> parameters)
+        {
+            var pooledBuilder = PooledStringBuilder.GetInstance();
+
+            var builder = pooledBuilder.Builder;
+            builder.Append(openBrace);
+            AppendParameters(parameters, builder);
+            builder.Append(closeBrace);
+
+            return pooledBuilder.ToStringAndFree();
+        }
+
+        private void AppendParameters(SeparatedSyntaxList<ParameterSyntax> parameters, StringBuilder builder)
+        {
+            var first = true;
+            foreach (var parameter in parameters)
+            {
+                if (!first)
+                {
+                    builder.Append(", ");
+                }
+
+                foreach (var modifier in parameter.Modifiers)
+                {
+                    builder.Append(modifier.Text);
+                    builder.Append(' ');
+                }
+
+                AppendTokens(parameter.Type, builder);
+
+                first = false;
+            }
+        }
+
+        private bool IsExtensionMethod(MethodDeclarationSyntax method)
+            => method.ParameterList.Parameters.Count > 0 &&
+               method.ParameterList.Parameters[0].Modifiers.Any(SyntaxKind.ThisKeyword);
+
+        private Accessibility GetAccessibility(SyntaxNode node, SyntaxTokenList modifiers)
+        {
+            var sawInternal = false;
+            foreach (var modifier in modifiers)
+            {
+                switch (modifier.Kind())
+                {
+                    case SyntaxKind.PublicKeyword: return Accessibility.Public;
+                    case SyntaxKind.PrivateKeyword: return Accessibility.Private;
+                    case SyntaxKind.ProtectedKeyword: return Accessibility.Protected;
+                    case SyntaxKind.InternalKeyword:
+                        sawInternal = true;
+                        continue;
+                }
+            }
+
+            if (sawInternal)
+            {
+                return Accessibility.Internal;
+            }
+
+            // No accessibility modifiers:
+            switch (node.Parent.Kind())
+            {
+                case SyntaxKind.ClassDeclaration:
+                case SyntaxKind.StructDeclaration:
+                    // Anything without modifiers is private if it's in a class/struct declaration.
+                    return Accessibility.Private;
```

Comment:
"only if it's nested".  Right :)   That's why i said "if it's in a class/struct declaration"